celery/canvas.py

```diff
@@ -629,7 +633,8 @@ def delay(self, *partial_args, **partial_kwargs):
         return self.apply_async(partial_args)
 
     def apply_async(self, partial_args=(), args=(), kwargs={}, task_id=None,
-                    **options):
+                    producer=None, publisher=None, connection=None,
+                    router=None, result_cls=None, **options):
```

Comment:
This fixes one regression added in https://github.com/celery/celery/pull/3244/

I thought these parameters were not being used and removed them, when in fact they were being removed from `options` which is later handed over.